File: Problem/KakaoCode 2018/src/Dolls.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Dolls {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        int K = Integer.parseInt(st.nextToken());

        st = new StringTokenizer(br.readLine());
        int[] an = new int[N];
        for(int i = 0; i < N; i++) {
            an[i] = Integer.parseInt(st.nextToken());
        }
        double minimum = 1e18;

        for(int i = K; i <= N; i++ ) {
            for(int j = 0; j <= N - i; j++) {
                double mean = 0;
                for(int k = 0; k < i; k++) {
                    mean += an[j+k];
                }
                mean /= i;

                double deviation = 0;
                for(int k =0; k < i; k++) {
                    deviation += (mean - an[j+k]) * (mean - an[j+k]);
                }
                deviation /= i;

                if(minimum > deviation) {
                    minimum = deviation;
                }
            }
        }

        System.out.printf("%.10f",Math.sqrt(minimum));
    }
}

File: Problem/KakaoCode 2018/src/Dolls.js
var fs = require('fs');
var inputData = fs.readFileSync('./dev/stdinB').toString().split(/\r\n|\r|\n/);

var N = parseInt(inputData[0].split(' ')[0]);
var K = parseInt(inputData[0].split(' ')[1]);

var an = inputData[1].split(' ');
for(var i = 0; i < N; i++) {
    an[i] = parseInt(an[i]);
}

var minimum = 1e18;

for(var i = K; i <= N; i++) {
    for(var j = 0; j <= N - i; j++) {
        var mean = 0;
        for(var k = 0; k < i; k++) {
            mean += an[j+k];
        }
        mean /= i;

        var deviation = 0;
        for(var k = 0; k < i; k++) {
            deviation += (mean - an[j+k]) * (mean - an[j+k]);
        }
        deviation /= i;
        if(minimum > deviation) {
            minimum = deviation;
        }
    }
}

console.log(Math.sqrt(minimum).toFixed(10));